src/Statistics.jl

```diff
@@ -28,21 +48,63 @@ The algorithm is further described in
 > doi: [10.1137/12086282X](https://doi.org/10.1137/12086282X),
 > arxiv: [1201.0925](https://arxiv.org/abs/1201.0925)
 """
-function mean(M::Manifold, x::AbstractVector, w::AbstractWeights = (n = length(x); Weights(ones(n), n)); x0 = x[1], kwargs...)
-    y = similar_result(M, mean, x0)
-    copyto!(y,x0)
-    return mean!(M, y, x, w; kwargs...)
-end
+mean
+
 @doc doc"""
-    mean!(M,y,x,w)
+    mean!(M::Manifold, y, x::AbstractVector; kwargs...)
+    mean!(M::Manifold, y, x::AbstractVector, w::AbstractWeights; kwargs...)
+    mean!(M::Manifold, y, x::AbstractVector, method::AbstractMethod; kwargs...)
+    mean!(M::Manifold, y, x::AbstractVector, w::AbstractWeights, method::AbstractMethod; kwargs...)
 
-computes the [`mean`](@ref) in-place in `y` where the initial value of `y` is
-the starting point of the algorithm.
+Compute the [`mean`](@ref) in-place in `y`.
 """
-function mean!(M::Manifold, y, x::AbstractVector, w::AbstractWeights = (n = length(x); Weights(ones(n), n));
-            stop_iter=100,
-            kwargs...
-        ) where {T}
+mean!
+
```

Comment:
This is quite a lot of cases, but I see you want to always make the method optional...still, can some of these cases be combined?

Reply:
Do you mean in the docstring or in the code?

Reply:
in the code. Document them is fine I think

Reply:
I'll double check but this was the only way I could work out to make it easy to add new methods and change defaults for manifolds easily, since optional arguments cannot be followed by required arguments. It's a lot of cases here but makes it easier and less error prone to extend. Assuming we keep the code as is, are you fine with how these are documented?

Reply:
Yes, documentation is fine I think.